x-pack/plugins/security_solution/server/search_strategy/security_solution/index.ts

```diff
@@ -37,7 +38,9 @@ export const securitySolutionSearchStrategyProvider = <T extends FactoryQueryTyp
   getSpaceId?: (request: KibanaRequest) => string,
   ruleDataClient?: IRuleDataClient | null
 ): ISearchStrategy<StrategyRequestType<T>, StrategyResponseType<T>> => {
-  const es = data.search.getSearchStrategy(ENHANCED_ES_SEARCH_STRATEGY);
+  const es = data.search.getSearchStrategy(
```

Comment:
why does the inferred return type no longer work here?

Reply:
We needed to enable using a type in between request and response, I called it `StrategyParseResponseType` which default to `IEsSearchResponse` because in my case the response getting into `parse()` and whatever parse() returns are 2 different things, couldn't just use `StrategyResponseType`. 

This enabled us to add some more logic to search_strategy, instead of doing that on frontend queries. 

How does this sound?